src/librustdoc/doctest.rs

```diff
@@ -118,6 +165,15 @@ pub(crate) fn run(options: RustdocOptions) -> Result<(), ErrorGuaranteed> {
     let externs = options.externs.clone();
     let json_unused_externs = options.json_unused_externs;
 
+    let temp_dir = match tempdir()
```

Comment:
Could your reuse the same output directory ?
https://github.com/rust-lang/rust/blob/7776678bef8ca178b63d4aa50f734d72f419757d/src/librustdoc/doctest.rs#L966-L968

Reply:
I'll make a new function to be called in both places.